File: Zipping and Unzipping files in .NET application/[C#]-Zipping and Unzipping files in .NET application/C#/SFXCreation.cs
using System;
using System.Windows.Forms;

namespace ArchiveManager
{
    public partial class SFXCreation : Form
    {
        public SFXCreation(string stubFileName, string sfxOutputFileName)
        {
            InitializeComponent();

            txtName.Text = stubFileName;
            txtSfx.Text = sfxOutputFileName;
        }

        /// <summary>
        /// Gets the desired Stub file name.
        /// </summary>
        public string StubFileName
        {
            get
            {
                return txtName.Text;
            }
            set
            {
                txtName.Text = value;
            }
        }

        /// <summary>
        /// Gets the desired SFX file name.
        /// </summary>
        public string SfxFileName
        {
            get
            {
                return txtSfx.Text;
            }
            set
            {
                txtSfx.Text = value;
            }
        }

        /// <summary>
        /// Handles the OK button's Click event.
        /// </summary>
        /// <param name="sender">The button object.</param>
        /// <param name="e">The event arguments.</param>
        private void btnOk_Click(object sender, EventArgs e)
        {
#if false
            // Check for invalid characters.
            if (StubFileName.Contains("/") || StubFileName.Contains("\\"))
            {
                MessageBox.Show(Messages.InvalidCharacters, this.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop);
                return;
            }

            if (SfxFileName.Contains("/") || SfxFileName.Contains("\\"))
            {
                MessageBox.Show(Messages.InvalidCharacters, this.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop);
                return;
            }
#endif

            this.DialogResult = DialogResult.OK;
        }

        private void btnStubFile_Click(object sender, EventArgs e)
        {
            try
            {
                OpenFileDialog dlg = new OpenFileDialog();
                dlg.Title = "Select an SFX stub file";
                dlg.FileName = txtName.Text;
                dlg.Filter = "All files|*.*";
                dlg.FilterIndex = 1;
                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    txtName.Text = dlg.FileName;
                }
            }
            catch (Exception exc)
            {
                Util.ShowError(exc);
            }
        }

        private void btnBrowseSfx_Click(object sender, EventArgs e)
        {
            try
            {
                OpenFileDialog dlg = new OpenFileDialog();
                dlg.Title = "Select an SFX file";
                dlg.FileName = txtSfx.Text;
                dlg.Filter = "All files|*.*";
                dlg.FilterIndex = 1;
                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    txtSfx.Text = dlg.FileName;
                }
            }
            catch (Exception exc)
            {
                Util.ShowError(exc);
            }
        }
    }
}

File: Zipping and Unzipping files in .NET application/[VB]-Zipping and Unzipping files in .NET application/VB/SFXCreation.vb
Namespace ArchiveManager
	Partial Public Class SFXCreation
		Inherits Form
		Public Sub New(ByVal stubFileName As String, ByVal sfxOutputFileName As String)
			InitializeComponent()

			txtName.Text = stubFileName
			txtSfx.Text = sfxOutputFileName
		End Sub

		''' <summary>
		''' Gets the desired Stub file name.
		''' </summary>
		Public Property StubFileName() As String
			Get
				Return txtName.Text
			End Get
			Set(ByVal value As String)
				txtName.Text = value
			End Set
		End Property

		''' <summary>
		''' Gets the desired SFX file name.
		''' </summary>
		Public Property SfxFileName() As String
			Get
				Return txtSfx.Text
			End Get
			Set(ByVal value As String)
				txtSfx.Text = value
			End Set
		End Property

		''' <summary>
		''' Handles the OK button's Click event.
		''' </summary>
		''' <param name="sender">The button object.</param>
		''' <param name="e">The event arguments.</param>
		Private Sub btnOk_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnOk.Click
#If false Then
			' Check for invalid characters.
			If StubFileName.Contains("/") OrElse StubFileName.Contains("\") Then
				MessageBox.Show(Messages.InvalidCharacters, Me.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop)
				Return
			End If

			If SfxFileName.Contains("/") OrElse SfxFileName.Contains("\") Then
				MessageBox.Show(Messages.InvalidCharacters, Me.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop)
				Return
			End If
#End If

			Me.DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub

		Private Sub btnStubFile_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnStubFile.Click
			Try
				Dim dlg As New OpenFileDialog()
				dlg.Title = "Select an SFX stub file"
				dlg.FileName = txtName.Text
				dlg.Filter = "All files|*.*"
				dlg.FilterIndex = 1
				If dlg.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
					txtName.Text = dlg.FileName
				End If
			Catch exc As Exception
				Util.ShowError(exc)
			End Try
		End Sub

		Private Sub btnBrowseSfx_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnBrowseSfx.Click
			Try
				Dim dlg As New OpenFileDialog()
				dlg.Title = "Select an SFX file"
				dlg.FileName = txtSfx.Text
				dlg.Filter = "All files|*.*"
				dlg.FilterIndex = 1
				If dlg.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
					txtSfx.Text = dlg.FileName
				End If
			Catch exc As Exception
				Util.ShowError(exc)
			End Try
		End Sub
	End Class
End Namespace